chainstate/tx-verifier/src/transaction_verifier/input_output_policy.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::chain::{OutputPurpose, Transaction, TxOutput};
+
+use super::error::ConnectTransactionError;
+
+/// Not all `OutputPurposes` can be used in a transaction.
+/// For example spending `ProduceBlockFromStake` and `StakePool` in a tx is not supported
```

Comment:
I would say "not allowed". It's not like we're gonna allow it in the future :smile:

Reply:
I think we will if the output type is DecommissionPool